runtime/vm/BytecodeInterpreter.hpp

```diff
@@ -2610,14 +2610,29 @@ class INTERPRETER_CLASS
 		_sp += (slotCount - 1);
 	}
 
-	/* java.lang.Thread: public static native Thread _currentThread(); */
+	/* java.lang.Thread: public static native Thread currentThread(); */
 	VMINLINE VM_BytecodeAction
 	inlThreadCurrentThread(REGISTER_ARGS_LIST)
 	{
 		returnObjectFromINL(REGISTER_ARGS, _currentThread->threadObject, 0);
 		return EXECUTE_BYTECODE;
 	}
 
+#if JAVA_SPEC_VERSION >= 19
+	/* java.lang.Thread: native void setCurrentThread(Thread thread); */
+	VMINLINE VM_BytecodeAction
+	inlThreadSetCurrentThread(REGISTER_ARGS_LIST)
+	{
+		j9object_t thread = ((j9object_t*)_sp)[0];
+		j9object_t receiverObject = ((j9object_t*)_sp)[1];
+		J9VMThread *targetThread = J9VMJAVALANGTHREAD_THREADREF(_currentThread, receiverObject);
+		Assert_VM_notNull(targetThread);
```

Comment:
This isn't really useful - we either need to assume the receiver thread is live, or do a proper liveness check (which involves synchronization).

Reply:
Based on the java code, the receiver is always a carrier thread, so I think we can probably just assume they are alive when this API is called. The API itself is package private to java.lang so there won't be arbitrary callers.

Reply:
I would replace the assertion with a comment based on the above conversation.